### DISCUSSION TITLE
Ask two questions,Thanks.

### DISCUSSION THREAD

**Discussion by yslhzj** [jertel/elastalert2#775] [Q&A] [answered]
1. Excuse me, can python3.6 still use elastalert2 normally?  Does it have to be python3.9?
2. How do I combine and use multiple rule types in elastalert2 ? Like frequency and blacklist. For example, for 1000 servers, I only want to focus on the 200 machines included in blacklist, and the ip address appears five times in an hour.

**Comment by ferozsalam** [comment, marked as answer]
> 1. Excuse me, can python3.6 still use elastalert2 normally?  Does it have to be python3.9?

It's likely that most functionality will work, but the best way of telling would be to run it and see – as this is looked after by volunteers, there's no guarantee of long-term support.

If the most recent version doesn't work, you will probably be able to run an older version of the package if you don't need Elasticsearch 8 support – I think the versions from late 2020/early 2021 should be fine.

>     2. How do I combine and use multiple rules in elastalert2 ? Like frequency and blacklist.

Could you describe exactly what you would be trying to alert on in this case? An example would be useful. It's not directly possible as far as I'm aware but there are a couple of workarounds that might do what you want.

**Reply by nsano-rururu** [reply]
By the way, as of 2021/12/23, Python 3.6 has become EOL (End Of Life).
From a security point of view, I think you should choose Python 3.7 or later.

**Reply by yslhzj** [reply]
Thank for your reply.


How do I combine and use multiple rule types in elastalert2 ? Like frequency and blacklist.

For example, for 1000 servers, I only want to focus on the 200 machines included in blacklist, and the ip address appears five times in an hour.


> > 2. How do I combine and use multiple rules in elastalert2 ? Like frequency and blacklist.
> > ```
> 
> Could you describe exactly what you would be trying to alert on in this case? An example would be useful. It's not directly possible as far as I'm aware but there are a couple of workarounds that might do what you want.

**Reply by nsano-rururu** [reply]
@yslhzj 

It is not supported by the standard function, so it cannot be done.
